include/seqan3/std/concepts

```diff
@@ -17,6 +17,66 @@
 
 #ifdef __cpp_lib_concepts        // C++20 ranges available
 #include <concepts>
+
```

Comment:
Here you haven't made the ifdef of including `<concepts>` separate from `__cpp_lib_concepts` as in the last PR I reviewed, I thought it is favourable?

Reply:
You are right and I'm also confused why this works;

EDIT:// boah `platform.hpp` includes `<version>` which defines `__cpp_lib_concepts` :facepalm:

Reply:
I'll change this to the new pattern.